app/src/main/java/org/dicekeys/app/fragments/ScanFragment.kt

```diff
@@ -0,0 +1,185 @@
+                findNavController().popBackStack()
+            }
+        }
+    }
+
+    /**
+     * Check if all permission specified in the manifest have been granted
+     */
+    private fun allPermissionsGranted() = REQUIRED_PERMISSIONS.all {
+        ContextCompat.checkSelfPermission(requireContext(), it) == PackageManager.PERMISSION_GRANTED
+    }
+
+    private fun startCamera() {
+        cameraProviderFuture.addListener(Runnable {
+            val cameraProvider = cameraProviderFuture.get()
+            bindPreview(cameraProvider)
+        }, ContextCompat.getMainExecutor(requireContext()))
+    }
+
+    private fun bindPreview(cameraProvider: ProcessCameraProvider) {
+        var cameraSelector: CameraSelector = CameraSelector.Builder()
+                .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+                .build()
+        //val previewSize = Size(viewFinder.width, viewFinder.height)
+        val previewSize = Size(binding.previewView.width, binding.previewView.height)
+
+
+        // Build the viewfinder use case
+        val preview: Preview = Preview.Builder()
+                .setTargetResolution(previewSize)
+                .build()
+
+        val pWidth = binding.previewView.width
+        val pHeight = binding.previewView.height
+        val analyzerSize: Size =
+                if (pWidth * 9 > pHeight * 16)
+                // Wider than 16x9, so fix width=1920 and calculate a height which will be <= 1080
+                    Size(1920, (1920 * pHeight) / pWidth)
+                else if (pHeight * 16 > pWidth * 9)
+                // Taller than 16x9, so fix height=1920 and calculate a width which will be <= 1080
+                    Size((1920 * pWidth) / pHeight, 1920)
+                else if (pWidth > pHeight)
+                // Wider than 1x1 but less than 16x9, so fix height=1080 and calculate width <=1920
+                    Size((1080 * pWidth / pHeight), 1080)
+                else
+                // Taller than 1x1, or 1x1, so fix width at 1080 and calculate height 1080<=x<=1920
+                    Size(1080, (1080 * pHeight / pWidth))
+        val diceKeyImageAnalyzerUseCase = ImageAnalysis.Builder()
+                // In our analysis, we care more about the latest image than
+                // analyzing *every* image
+                .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+                .setTargetResolution(analyzerSize)
+                .build()
+
+        var analyzeDiceKey = DiceKeyAnalyzer(requireActivity())
+        diceKeyImageAnalyzerUseCase.setAnalyzer(
+                // Stuart is guessing with this next parameter
+                executor, // ContextCompat.getMainExecutor(this),
+                analyzeDiceKey
+        )
+        // analyzerConfig.setTargetRotation()
+
+        analyzeDiceKey.onActionOverlay = fun(overlayBitmap) {
+            val matrix = Matrix()
+            // FIXME - not sure this will be correct if scanning in landscape.
```

Comment:
An old FIXME.  Worth investigating if it's still an issue we should create an issue for or if we can remove the note.